### PR TITLE
Misc small changes and fixes

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Some preparations for paged secure shared memory and some random stuff.

### CODE REVIEW

**Comment by jforissier** [comment]
All commits except "core: bugfix core_mmu_get_entry_primitive" (which I'm not knowledgeable about):
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

BTW you should always add parentheses to function names in commit subjects/descriptions, so that they better stand out as functions.

**Comment by jenswi-linaro** [comment]
OK, I'll add some parentheses when updating the commits. By the way "core: bugfix core_mmu_get_entry_primitive" can be crosschecked against:
Figure B3-4 Short-descriptor first-level descriptor formats
and
Figure B3-15 Long-descriptor third-level descriptor formats
in the ARMv7-A ARM ARM

File: core/arch/arm/include/mm/tee_pager.h [hunk 1]
```diff
@@ -101,8 +101,8 @@ bool tee_pager_add_uta_area(struct user_ta_ctx *utc, vaddr_t base, size_t size);
  *
  * Return true on success of false if the area can't be updated
  */
-bool tee_pager_set_uta_area(struct user_ta_ctx *utc, vaddr_t base, size_t size,
```
Comment (etienne-lms) [inline core/arch/arm/include/mm/tee_pager.h]: also change the 2 comment above:
```
- * finalized with tee_pager_set_uta_area() to get more strict settings.
+ * finalized with tee_pager_set_uta_area_attr() to get more strict settings.
```
```
- * tee_pager_set_uta_area() - Set attributes of a initialized memory area
+ * tee_pager_set_uta_area_attr() - Set attributes of a initialized memory area
```

File: core/arch/arm/mm/core_mmu_lpae.c [hunk 2]
```diff
@@ -204,6 +204,14 @@ static uint32_t desc_to_mattr(uint64_t desc)
 		return 0;
 	}
 
+	if (level == 3) {
+		if ((desc & L3_BLOCK_DESC) != L3_BLOCK_DESC)
```
Comment (etienne-lms) [inline core/arch/arm/mm/core_mmu_lpae.c]: minor: it's a bit weird to use `L3_BLOCK_DESC` as both mask and value. same below.
(note that this was already done in `core_mmu_find_table()`). 
An `DESC_ENTRY_TYPE_MASK` (or so) could be more... explicit.

File: core/arch/arm/mm/core_mmu_v7.c [hunk 3]
```diff
@@ -413,8 +421,7 @@ void core_mmu_create_user_map(struct user_ta_ctx *utc,
 	COMPILE_TIME_ASSERT(sizeof(uint32_t) * TEE_MMU_L2_NUM_ENTRIES ==
 			    PGT_SIZE);
 
-	core_mmu_set_info_table(&dir_info, 1, 0, tbl);
-	dir_info.num_entries = TEE_MMU_UL1_NUM_ENTRIES;
+	core_mmu_get_user_pgdir(&dir_info);
 	memset(tbl, 0, sizeof(uint32_t) * TEE_MMU_UL1_NUM_ENTRIES);
```
Comment (etienne-lms) [inline core/arch/arm/mm/core_mmu_v7.c]: minor: for consistency with the lpae implementation:
```
- memset(tbl, 0, sizeof(uint32_t) * TEE_MMU_UL1_NUM_ENTRIES);
+ memset(dir_info.table, 0, dir_info.num_entries * sizeof(uint32_t));
```

File: core/arch/arm/mm/tee_pager.c [hunk 4]
```diff
@@ -633,6 +576,75 @@ static vaddr_t __maybe_unused area_idx2va(struct tee_pager_area *area,
...
+			return false;
+		TAILQ_INIT(utc->areas);
+	}
+
+	flags = TEE_MATTR_PRW | TEE_MATTR_URWX;
+
+	while (s) {
+		size_t s2;
+
+		if (find_area(utc->areas, b))
```
Comment (etienne-lms) [inline core/arch/arm/mm/tee_pager.c]: private question (change is ok): why did you need to add this check ?

Comment (jenswi-linaro) [inline core/arch/arm/mm/tee_pager.c]: Just to catch an eventual conflict when `tee_pager_add_uta_area()` later is used to add areas in a more dynamic way. A conflict here would be quite tricky to find in other ways.

Comment (etienne-lms) [inline core/arch/arm/mm/tee_pager.c]: ok, thanks.

**Comment by jenswi-linaro** [comment]
Please wait, I need to update the commit messages

**Comment by jenswi-linaro** [comment]
Added parentheses after function names in the commit messages as requested
Comments addressed and tags applied, ready to merge.

**Comment by jenswi-linaro** [comment]
Rebased on master
